contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/util/SimpleHttp.java

```diff
@@ -0,0 +1,240 @@
+    builder.writeTimeout(config.timeout(), TimeUnit.SECONDS);
+    builder.readTimeout(config.timeout(), TimeUnit.SECONDS);
+
+    return builder.build();
+  }
+
+  /**
+   * This function accepts a Builder object as input and configures response caching. In order for
+   * caching to work, the DRILL_TMP_DIR variable must be set either as a system environment variable or in the
+   * Drill configurations.
+   * <p>
+   * The function will attempt to get the DRILL_TMP_DIR from these places, and if it cannot, it will issue a warning in the logger.
+   *
+   * @param builder Builder the Builder object to which the cacheing is to be configured
+   */
+  private void setupCache(Builder builder) {
+    int cacheSize = 10 * 1024 * 1024;   // TODO Add cache size in MB to config
+    String drillTempDir;
+
+    try {
+      if (context.getOptions().getOption(ExecConstants.DRILL_TMP_DIR) != null) {
+        drillTempDir = context.getOptions().getOption(ExecConstants.DRILL_TMP_DIR).string_val;
+      } else {
+        drillTempDir = System.getenv("DRILL_TMP_DIR");
+      }
+      File cacheDirectory = new File(drillTempDir);
+      if (cacheDirectory == null) {
+        logger.warn("HTTP Storage plugin caching requires the DRILL_TMP_DIR to be configured. Please either set DRILL_TMP_DIR or disable HTTP caching.");
+      } else {
+        Cache cache = new Cache(cacheDirectory, cacheSize);
+        logger.debug("Caching HTTP Query Results at: {}", drillTempDir);
+
+        builder.cache(cache);
+      }
+    } catch (Exception e) {
+      logger.warn("HTTP Storage plugin caching requires the DRILL_TMP_DIR to be configured. Please either set DRILL_TMP_DIR or disable HTTP caching.");
+    }
+  }
+
+  /**
+   * This function accepts text from a post body in the format:
+   * key1=value1
+   * key2=value2
+   *
+   * and creates the appropriate headers.
+   *
+   * @return FormBodu.Builder The populated formbody builder
+   */
+  private FormBody.Builder buildPostBody() {
+    final Pattern postBpdyPattern = Pattern.compile("^.+=.+$");
+
+    FormBody.Builder formBodyBuilder = new FormBody.Builder();
+    String[] lines = apiConfig.postBody().split("\\r?\\n");
+    for(String line : lines) {
+
+      // If the string is in the format key=value split it,
+      // Otherwise ignore
+      if (postBpdyPattern.matcher(line).find()) {
+        //Split into key/value
+        String[] parts = line.split("=");
+        formBodyBuilder.add(parts[0], parts[1]);
+      }
+    }
+    return formBodyBuilder;
+  }
+
+  /**
+   * This class intercepts requests and adds authentication headers to the request
+   */
+  public static class BasicAuthInterceptor implements Interceptor {
+    private final String credentials;
+
+    public BasicAuthInterceptor(String user, String password) {
+      credentials = Credentials.basic(user, password);
+      logger.debug("Intercepting request adding creds: {}", credentials);

```

Comment:
Please do not log credentials. Security guys (such as yourself) don't like that.

Reply:
Fixed...